src/probnum/_probabilistic_numerical_method.py

```diff
@@ -0,0 +1,53 @@
+"""Probabilistic Numerical Methods."""
+
+from abc import ABC, abstractmethod
+from typing import Generic, Tuple, TypeVar
+
+ProblemType = TypeVar("ProblemType")
+BeliefType = TypeVar("BeliefType")
+StateType = TypeVar("StateType")
+
+
+class ProbabilisticNumericalMethod(ABC, Generic[ProblemType, BeliefType]):
+    """Probabilistic numerical methods.
+
+    An abstract base class defining the implementation of a probabilistic numerical
+    method [1]_ [2]_. A PN method solves a numerical problem by treating it as a
+    probabilistic inference task.
+
+    Parameters
+    ----------
+    prior :
+        Prior knowledge about quantities of interest for the numerical problem.
+
+    References
+    ----------
+    .. [1] Hennig, P., Osborne, Mike A. and Girolami M., Probabilistic numerics and
+       uncertainty in computations. *Proceedings of the Royal Society of London A:
+       Mathematical, Physical and Engineering Sciences*, 471(2179), 2015.
+    .. [2] Cockayne, J., Oates, C., Sullivan Tim J. and Girolami M., Bayesian
+       probabilistic numerical methods. *SIAM Review*, 61(4):756–789, 2019
+
+    Notes
+    -----
+    All PN methods should subclass this base class. Typically convenience functions
+    (such as :meth:`~probnum.linalg.problinsolve`) will instantiate an object of a
+    derived subclass.
+    """
+
+    def __init__(self, prior: BeliefType):
+        self.prior = prior
+
+    @abstractmethod
+    def solve(
```

Comment:
what happened to `infer_solution`? :) I thought it had some elegance to it. Any particular reason you went back to solve? (If I remember correctly, we decided against solve back in winter, because it sounds a bit weird to solve a quadrature problem.) Also, `infer_solution` also teaches a user about PN, just from the function name! :)

Reply:
I realize this is what your comment says in #280, but I actually thought that we agreed on solve in the verbal discussion afterwards :grin:, because in particular for linear systems you don't always just infer the solution but also other quantities of interest (i.e. inverse, rhs, etc.). Also for quadrature I find "solving an integral/integration problem/quadrature problem" pretty natural.

As with all things naming, how about we finish the PR, put it on the agenda for the software meeting and then rename the method, once we all agree?

Reply:
Sure, that is fine by me. I was just wondering :)